src/main/java/com/palantir/docker/compose/DockerComposition.java

```diff
@@ -108,4 +110,25 @@ public static DockerCompositionBuilder of(DockerCompose executable) {
         return new DockerCompositionBuilder(executable);
     }
 
+    public static DockerCompositionBuilder fromService(DockerService service) {
+        return fromServiceCluster(ServiceCluster.of(service));
+    }
+
+    public static DockerCompositionBuilder fromService(DockerService service, DockerMachine dockerMachine) {
+        return fromServiceCluster(ServiceCluster.of(service), dockerMachine);
+    }
+
+    public static DockerCompositionBuilder fromServiceCluster(ServiceCluster services) {
+        return fromServiceCluster(services, DockerMachine.localMachine().build());
+    }
+
+    public static DockerCompositionBuilder fromServiceCluster(ServiceCluster services, DockerMachine dockerMachine) {
+        return fromServiceCluster(new DockerCompose(services.dockerComposeFiles(), dockerMachine), services);
+    }
+
+    public static DockerCompositionBuilder fromServiceCluster(DockerCompose executable, ServiceCluster services) {
+        DockerCompositionBuilder builder = new DockerCompositionBuilder(executable);
+        services.addToComposition(builder);
```

Comment:
Yeah, this is the bit that's making me uncomfortable - when Service is just a thing that exposes it's DockerComposeFile and a HealthCheck then this looks like a solid change. 

You can pass around snippets of docker stuff with how you wait for it and inject them in, which I like.